src/components/MultiSelect/MultiSelectMenuButton.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import Button from "./../Button/Button";
+import Icon from "./../Icons/Icon";
+import { Box, useMultiStyleConfig } from "@chakra-ui/react";
+import { SelectedItems } from "./MultiSelectTypes";
+
+export interface MultiSelectMenuButtonProps {
+  /** The id of the multiSelect using this button. */
+  multiSelectId: string;
+  /** The label of the multiSelect using this button. */
+  label: string;
+  /** The open status of the multiselect menu. */
+  isOpen: boolean;
+  /** The selected items state (items that were checked by user). */
+  selectedItems: SelectedItems;
+  /** The callback function for the menu toggle. */
+  onMenuToggle?: () => void;
+  /** The action to perform for clear/reset button of multiselect. */
+  onClear?: () => void;
+}
+
+const MultiSelectMenuButton = React.forwardRef<
+  HTMLButtonElement,
+  MultiSelectMenuButtonProps
+>((props, ref?) => {
+  const { multiSelectId, label, isOpen, onMenuToggle, selectedItems, onClear } =
+    props;
+  const styles = useMultiStyleConfig("MultiSelect", {});
+  const iconType = isOpen ? "minus" : "plus";
+
+  // Sets the selected items count on the menu button.
+  function getSelectedItemsCount(multiSelectId: string) {
+    if (selectedItems[multiSelectId]?.items.length > 0) {
+      return `${selectedItems[multiSelectId].items.length}`;
+    }
+    return null;
+  }
+
+  // We need this for our "fake" button inside the main menu button.
+  function onKeyPress(e) {
+    const enterOrSpace =
+      e.key === "Enter" ||
+      e.key === " " ||
+      e.key === "Spacebar" ||
+      e.which === 13 ||
+      e.which === 32;
+
+    if (enterOrSpace) {
+      e.preventDefault();
+      onClear();
+    }
+  }
+
+  return (
+    <Button
+      id={`${multiSelectId}-button`}
+      ref={ref}
+      __css={styles.menuButton}
+      buttonType="secondary"
+      onClick={onMenuToggle}
+      {...props}
```

Comment:
I think this is a typo and causes warnings in the console:
<img width="1326" alt="Screen Shot 2022-06-03 at 12 23 09 PM" src="https://user-images.githubusercontent.com/1280564/171906105-229284f2-8666-4431-bed2-c67921ecc931.png">

This happens because all the props are added to `Button` but the `<button>` HTML element doesn't know what the props passed are. For example, from the screenshot, it doesn't know what to do when it renders `<button multiSelectId="..." >`. You can safely remove this line here since you also pass the necessary props to its appropriate value. For example, `onMenuToggle` prop is passed to `onClick`.

Reply:
fixed.